### PR TITLE
Clean up `rcell` and unused argument documentation

### LINKED ISSUE #505: rcell function documentation
https://github.com/Roche/rtables/blob/231ffa0a1076d57c4f12e8b44975d083cef397ee/man/rcell.Rd#L34

What is the `FUN` exactly?

### PR DESCRIPTION
**Pull request by edelarua**
Cleaned up documentation for `rcell` function and rewrote documentation for `format` argument.

Removed `compat_args` documentation for arguments that are no longer used (`FUN`, `col_by`, `row_by`, and `col_wise_args`).

Closes #505

### CODE REVIEW

File: R/argument_conventions.R [hunk 1]
```diff
@@ -231,31 +231,20 @@ constr_args <- function(kids, cont, lev, iscontent, cinfo, labelrow, vals,
...
+#'   format labels.
 #' @param indent deprecated.
-#' @param col_wise_args a named list containing collections (e.g. vectors or
-#'   lists) with data elements for each column of the resulting table. The data
-#'   elements are then passed to the named argument \code{FUN} corresponding to
-#'   the element name of the outer list. Hence, the length and order of each
-#'   collection must match the levels in \code{col_by}. See examples.
 #' @param inset integer(1). The table inset for the row or table being
 #'   constructed. See \code{\link[formatters]{table_inset}}.
 #' @rdname compat_args
```
Comment (Melkiades) [inline R/argument_conventions.R:246]: I was wondering if we could add a more detailed description of each of these parameter categories. Should we also add `@keywords internal` to this function family?

**Review by Melkiades — APPROVED** [review]
Lgtm! I have just a minor doubt about possible improvements to these argument conventions

**Comment by shajoezhu** [comment]
This looks good to me! good to go

File: R/argument_conventions.R (hunk 1 above)
Comment (shajoezhu) [reply R/argument_conventions.R:246]: Hi @Melkiades , i think that is ok, if you think this adds value, can create issue and implement seperately

**Pull request merged**